### ISSUE TITLE
`test -t 1` returns true when not attached to tty

### ISSUE THREAD

**Issue by zx8** [fish-shell/fish-shell#4766]
**Fish**
```
$ fish --version
fish, version 2.7.1-836-g6f1ae79a

$ function is_attached_to_terminal; test -t 1; and echo yes; or echo no; end

$ echo | is_attached_to_terminal | cat
yes
```

**Bash**
```
$ bash --version
GNU bash, version 4.4.19(1)-release (x86_64-apple-darwin15.6.0)

$ is_attached_to_terminal() { test -t 1 && echo yes || echo no; }

$ echo | is_attached_to_terminal | cat
no
```

**Comment by faho** [comment]
Yup. Duplicate of #1228. Either use `isatty` or `command test`.

**Comment by faho** [comment]
Okay, I'm going to reopen this one and then fix it separately from #1228.

The reason for that is that I'm not going to make `test` *actually* redirect these file descriptors, because that's a complete waste.

Instead, I'm going to have test check what you *mean*, which is if the stream goes to a terminal even if the fd doesn't.

And that's a much bigger problem with the actual device _files_ because there are a bunch of them and I don't want to parse them out. To a first approximation, nobody cares what `test -c /dev/fd/0` does. They care if stdin is a tty, and `test -t 0` or `isatty stdin` are the thing to check.

If anyone cares enough to fiddle with `/dev/fd/x` and `/dev/stdout` and such, be my guest.
